File: Java/VSCode/src/MatricesMaquinaExpendedora.java
/*Estás haciendo un programa para una máquina expendedora de bebidas. El menú de las bebidas se almacena en una matriz llamada menu: 
String[] menu = {"Tea", "Espresso", "Americano", "Water", "Hot Chocolate"}; 
Toma la elección del cliente como un número entero de entrada y genera el elemento de menú correspondiente.
Además, comprueba si hay errores: en caso de que la entrada esté fuera del rango de la matriz, genera "Invalid". */

import java.util.Scanner;

public class MatricesMaquinaExpendedora {
    public static void main(String[] args) {
        
        System.out.println("Seleccionar una bebida: ");
        System.out.println("1. Tea\n2. Espresso\n3. Americano\n4. Water\n5. Hot Chocolate");

        String[] menu = {"Tea", "Espresso", "Americano", "Water", "Hot Chocolate"};

        // Usar try-with-resources para manejar el Scanner y cerrarlo automáticamente
        try (Scanner sc = new Scanner(System.in)) {
            int seleccion = sc.nextInt();

            // Verificar si la selección está en el rango válido (1 a 5)
            if (seleccion >= 1 && seleccion <= 5) {
                System.out.println(menu[seleccion - 1]); // Ajustar índice para el array
            } else {
                System.out.println("Invalido");
            }
        } // No se necesita cerrar manualmente el Scanner; se cierra automáticamente al final del bloque try

    }//End program

}//End class

File: Python/MatricesMaquinaExpendedora.py
"""
Instrucciones:
Programa para una máquina expendedora de bebidas que:
- Muestra un menú de bebidas disponibles
- Acepta la selección del usuario como número
- Muestra la bebida seleccionada
- Valida que la selección sea válida
- Muestra "Invalido" si la selección está fuera de rango o si hay una entrada errónea
"""

def mostrar_menu():
    print("Seleccionar una bebida: ")
    print("1. Tea")
    print("2. Espresso")
    print("3. Americano")
    print("4. Water")
    print("5. Hot Chocolate")

def main():
    menu = ["Tea", "Espresso", "Americano", "Water", "Hot Chocolate"]
    
    while True:
        mostrar_menu()
        try:
            seleccion = int(input("Introduce el número de tu selección (1-5): "))
            if 1 <= seleccion <= 5:
                print(f"Has seleccionado: {menu[seleccion - 1]}")
                break  # Sale del bucle si la selección es válida
            else:
                print("Invalido: Por favor elige un número entre 1 y 5.")
        except ValueError:
            print("Invalido: Por favor introduce un número entero.")

if __name__ == "__main__":
    main()
